src/server/groups/omegaMembershipGroups/read.ts

```diff
@@ -1,15 +1,58 @@
 import { prismaCall } from '@/server/prismaCall'
 import prisma from '@/prisma'
+import { ServerError } from '@/server/error'
+import type { OmegaMembershipLevel } from '@prisma/client'
 import type { ExpandedOmegaMembershipGroup } from './Types'
 
 export async function readOmegaMembershipGroups(): Promise<ExpandedOmegaMembershipGroup[]> {
     return await prismaCall(() => prisma.omegaMembershipGroup.findMany())
 }
 
-export async function readOmegaMembershipGroup(id: number): Promise<ExpandedOmegaMembershipGroup> {
+export async function readOmegaMembershipGroup(id: number | OmegaMembershipLevel): Promise<ExpandedOmegaMembershipGroup> {
+    if (typeof id === 'number') {
+        return await prismaCall(() => prisma.omegaMembershipGroup.findUniqueOrThrow({
+            where: {
+                id,
+            },
+        }))
+    }
+
     return await prismaCall(() => prisma.omegaMembershipGroup.findUniqueOrThrow({
         where: {
-            id,
+            omegaMembershipLevel: id,
+        },
+    }))
+}
+
+export async function readUserOmegaMembershipLevel(userId: number): Promise<OmegaMembershipLevel> {
+    const results = await prismaCall(() => prisma.user.findUniqueOrThrow({
+        where: {
+            id: userId,
         },
+        select: {
+            memberships: {
+                select: {
+                    group: {
+                        select: {
+                            omegaMembershipGroup: {
+                                select: {
+                                    omegaMembershipLevel: true,
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
     }))
+
+    for (let i = 0; i < results.memberships.length; i++) {
+        const g = results.memberships[i]
```

Comment:
wouldn't a more efficient approach be to get the ids of the omega membership groups

`const groupIds = prisma.omegaMembershipGroup.findMany({ select: { id: true } })`

and then check if there exists a membership between the user and any of the groups?

`prisma.memberships.findUnique({ where: { userId, groupId: { OR: groupIds } } })`